src/huggingface_hub/repository.py

```diff
@@ -869,14 +869,118 @@ def git_checkout(self, revision, create_branch_ok=False):
                 except subprocess.CalledProcessError as exc:
                     raise EnvironmentError(exc.stderr)
 
-    def push_to_hub(self, commit_message: str = "commit files to HF hub") -> str:
+    def tag_exists(self, tag_name: str) -> bool:
+        """
+        Check if a tag exists or not
+        """
+        try:
+            git_tags = subprocess.run(
+                ["git", "tag"],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            ).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        git_tags = git_tags.split("\n")
+        return tag_name in git_tags
+
+    def delete_tag(self, tag_name: str) -> bool:
+        """
+        Delete a tag, both local and remote, if it exists
+
+        Return True if deleted.  Returns False if the tag didn't exist
+        """
+        if not self.tag_exists(tag_name):
+            return False
+
+        try:
+            git_tags = subprocess.run(
+                ["git", "tag", "-d", tag_name],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            ).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        try:
+            git_tags = subprocess.run(
+                ["git", "push", "origin", "--delete", tag_name],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            ).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        return True
+
+    def add_tag(self, tag_name: str, message: str):
+        """
+        Add a tag at the current head and push it
```

Comment:
git has two types of tags: `lightweight` and `annotated`. Do we also want to support lightweight tags (defaulting to annotated?) for completeness? Or we don't care?

Reply:
I didn't care too much, but I could see renaming the method `add_annotated_tag` or just leaving message None to indicate lightweight

Reply:
I think having `message` as an optional string (lightweight when `None` and annotated when a message is specified would be super nice!

Reply:
Sure